hydra-node/src/Hydra/Chain/Direct/State.hs

```diff
@@ -159,6 +159,10 @@ instance IsChainState Tx where
   chainStateSlot ChainStateAt{recordedAt} =
     maybe (ChainSlot 0) chainSlotFromPoint recordedAt
 
+  findBy ChainStateAt{spendableUTxO} p =
+    uncurry UTxO.singleton
+      <$> UTxO.findBy (\(TxIn txId _, out) -> p txId out) spendableUTxO
```

Comment:
Interesting idea of finding the `depositTxId` by looking into the known (spendable) UTxO of the chain backend. However, this is very hacky and only works because we use the `depositTxId` as the sole piece of information to identify a deposit.